packages/craco/lib/features/test/jest-babel-transform.js

```diff
@@ -1,38 +1,26 @@
-const babelJest = require("babel-jest");
-
 const { loadCracoConfig } = require("../../config");
-const { isArray } = require("../../utils");
-
-const craBabelTransformer = {
-    presets: ["babel-preset-react-app"],
-    babelrc: false,
-    configFile: false
-};
-
-const context = {
-    env: process.env.NODE_ENV
-};
-
-const cracoConfig = loadCracoConfig(context);
-
-const { addPresets, addPlugins } = cracoConfig.jest.babel;
-
-if (cracoConfig.babel) {
-    if (addPresets) {
-        const { presets } = cracoConfig.babel;
-
-        if (isArray(presets)) {
-            craBabelTransformer.presets = craBabelTransformer.presets.concat(presets);
+const { createJestBabelTransform } = require("./create-jest-babel-transform");
+
+let jestBabelTransform;
+
+// cracoConfig is only available inside the transform, but the transform needs to include whatever options cracoConfig
+// specifies. So, the first time this transform is run, it generates a new transform -- using cracoConfig -- and
+// uses that to process files.
+module.exports = {
+    ...createJestBabelTransform(),
```

Comment:
What do you need this for? It is also called without a craco config. Is it why you also load the craco config inside the transformer?

Reply:
This populates any other keys that `babelJest.createTransformer` generates: in addition to `process()`, they make a [`getCacheKey` function](https://github.com/facebook/jest/blob/master/packages/babel-jest/src/index.ts#L97) and set a `canInstrument` prop. That's the only thing it's used for.

We could set all keys directly here -- it just seems possibly more future-proof to start with whatever babel-jest creates and override just the `process()`. I can update it if you'd prefer. Or, for about the same work we could probably override all functions (i.e. have `getCacheKey` do the same "generate and use jestBabelTransform from the config" that `process` does), although in my tests it seemed unnecessary.